index.ts

```diff
@@ -1,8 +1,16 @@
 import validator from "validator";
-import { EventSource } from "undici";
+import {
+  EventSource,
+  EnvHttpProxyAgent,
+  setGlobalDispatcher,
+  type ErrorEvent,
+  type MessageEvent,
+} from "undici";
 import url from "url";
 import querystring from "querystring";
 
+setGlobalDispatcher(new EnvHttpProxyAgent());
```

Comment:
actually EventSource and fetch accept dispatcher parameters or so.

Reply:
Doing so required the use of undici's fetch since the types were incompatible
9b827fb1d74bb8ea05bb34099958fe826a628acf